cmake/helper_functions.cmake

```diff
@@ -0,0 +1,70 @@
+# ign_generate_conf()
+# Generate conf/*.yaml files for projects
```

Comment:
Knit: can we name this file something different. "helper_functions is not very descriptive.

Reply:
How about ign_generate_files, we can keep other macros that generate boilerplate under this later.

Reply:
I have changed the name to configuration_functions as discussed.